sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -112,6 +114,34 @@ class CacheManager extends Logging with AdaptiveSparkPlanHelper {
     }
   }
 
+  /**
+   * Un-cache all cache entries that refer to the given table.
+   *
+   * @param table The [[Table]] to be un-cached
+   */
+  def uncacheV2Table(table: Table): Unit = {
```

Comment:
can we avoid changing `CacheManager`? We shouldn't put data source specific logic in `CacheManager`, which should be general and also handle logical plans.